### PR TITLE
Add tolerance to early stopping.

### PR DESCRIPTION
**Pull request by trivialfis**
Close https://github.com/dmlc/xgboost/issues/4982 .

### CODE REVIEW

**Comment by candalfigomoro** [comment]
Can this also be added to XGBoost4J and XGBoost4J-Spark?

**Comment by trivialfis** [comment]
@candalfigomoro I'm not sure, @wbo4958 has some interest in better es support in JVM packages, but that will depend on his priority.

File: python-package/xgboost/callback.py [hunk 1]
```diff
@@ -487,25 +487,25 @@ class EarlyStopping(TrainingCallback):
...
     """
     def __init__(self,
                  rounds: int,
                  metric_name: Optional[str] = None,
                  data_name: Optional[str] = None,
                  maximize: Optional[bool] = None,
-                 save_best: Optional[bool] = False) -> None:
+                 save_best: Optional[bool] = False,
+                 tolerance: float = 0) -> None:
         self.data = data_name
```
[the remark this answers is not in the document]
Comment (trivialfis) [reply python-package/xgboost/callback.py:517]: Changed the name to `abs_tol` with documents.  I don't think relative tolerance is applicable here.  Relatively greater than?

**Review by RAMitchell — APPROVED** [review]

**Comment by MJimitater** [comment]
Great work! Is this also added in XGBoost on R Cran?

**Pull request merged**